### ISSUE TITLE
Kotlin 3.1.6, SharedViewModelCompat.sharedViewModel throw exception

### ISSUE THREAD

**Issue by LordKered** [InsertKoinIO/koin#1332]
Hello. We are using for some Java Fragments SharedViewModelCompat; After updating got an issue.
Old Version (work): 3.1.5
New Version (Issue): 3.1.6
**Describe the bug**
`Lazy<AnyViewModel> viewModel = SharedViewModelCompat.sharedViewModel(this, AnyViewModel.class);`
Will thow` IllegalStateException("Fragment " + this + " not attached to an activity."). `
As I see in Kotlin 3.1.6 you move the lazy method inside `ViewModelCompat.viewModel`, but `fragment.requireActivity()` will call when the fragment is created (init process), activity still not attached. Is it issue or maybe you have any new way for this one?

**To Reproduce**
```
public class JavaFragment extends Fragment {

    private final Lazy<WeatherViewModel> viewModel = SharedViewModelCompat.sharedViewModel(this, WeatherViewModel.class);

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        //...
    }
}
```

**Comment by arnaudgiuliani** [comment]
The best here, is to update the API to use a Lazy value for the owner: 

```kotlin
ViewModelCompat.viewModel(lazy(LazyThreadSafetyMode.NONE) { fragment.requireActivity() }, clazz, qualifier, parameters)
```

The ViewModel Compat API become then:
```kotlin
fun <T : ViewModel> viewModel(
        owner: Lazy<ViewModelStoreOwner>,
        clazz: Class<T>,
        qualifier: Qualifier? = null,
        parameters: ParametersDefinition? = null
    )
```

**Comment by arnaudgiuliani** [comment]
Fixed in 2b8c1c97d336ab70697921910e9752eacc7ad2fa
